test/cli/whole-program/configs_unused.c

```diff
@@ -0,0 +1,18 @@
+// cppcheck-suppress misra-c2012-2.3
+typedef struct {
+    int x; // cppcheck-suppress unusedStructMember
```

Comment:
"unusedStructMember" is not a whole program analysis checker. so I think putting that in this whole program test is suspicious.

Reply:
I think  `unusedStructMember`, `unusedVariable`, etc. _should_ be checked against different configurations, similar to `unusedFunction`. 
However this was not main point here. -  For lack of better example, I used structure typedef to exercise one of the misra's system level rules that I was familiar with i.e. _Rule 2.3 - a project should not contain unused type declarations._

Reply:
I agree that we should ideally check these against different configurations but not sure if I think this is the proper test for that. I prefer short tests that are more to the point.

Reply:
I imagine that most "style" checkers would ideally look at all the configurations. A "known condition result" checker can also generate false positives for instance.